src/twister2/yaml_test_specification.py

```diff
@@ -55,3 +64,101 @@ def __post_init__(self):
         self.extra_args = string_to_list(self.extra_args)
         self.integration_platforms = string_to_list(self.integration_platforms)
         self.timeout *= self.timeout_multiplier
+
+
+# Using marshmallow to validate specification from yaml
+class HarnessConfigSchema(Schema):
+    type = fields.Str()
+    fixture = fields.Str()
+    ordered = fields.Str()
+    repeat = fields.Str()
+    pytest_root = fields.Str()
+    pytest_args = fields.List(fields.Str())
+    regex = fields.List(fields.Str())
+    record = fields.Dict(fields.Str(), fields.Str())
+
+
+class CommonSchema(Schema):
+    arch_exclude = fields.Str()
+    arch_allow = fields.Str()
+    testcases = fields.List(fields.Str)
+    build_only = fields.Bool()
+    build_on_all = fields.Bool()
+    depends_on = fields.Str()
+    extra_args = fields.Str()
+    extra_configs = fields.List(fields.Str())
+    extra_sections = fields.Str()
+    filter = fields.Str()
+    integration_platforms = fields.List(fields.Str())
+    harness = fields.Str()
+    harness_config = fields.Nested(HarnessConfigSchema())
+    min_ram = fields.Int()
+    min_flash = fields.Int()
+    modules = fields.List(fields.Str())
+    platform_exclude = fields.Str()
+    platform_allow = fields.Str()
+    platform_type = fields.List(fields.Str())
+    tags = fields.Str()
+    timeout = fields.Int()
+    toolchain_exclude = fields.Str()
+    toolchain_allow = fields.Str()
+    type = fields.Str(validate=validate.OneOf(['unit']))
+    skip = fields.Bool()
+    slow = fields.Bool()
+    sysbuild = fields.Bool()
+
+
+class SampleSchema(Schema):
+    name = fields.Str(required=True)
+    description = fields.Str()
+
+
+class TestsSchema(Schema):
```

Comment:
Maybe better name for this class is `TestSchema` (without "s")? It concerns particular one test not all tests.

Reply:
Changed to `TestSchema`